frontend/ui/impl/src/test/java/com/virtuslab/gitmachete/frontend/ui/impl/GitMacheteErrorReportSubmitterTest.java

```diff
@@ -0,0 +1,123 @@
+  // PowerMock API documentation Section 13 [https://javadoc.io/static/org.mockito/mockito-core/4.6.1/org/mockito/Mockito.html]
+  private static final GitMacheteErrorReportSubmitter reportSubmitter = PowerMockito
+      .spy(new GitMacheteErrorReportSubmitter());
+  private static final LambdaLogger errorReportSubmitterLogger = PowerMockito.mock(LambdaLogger.class);
+  private static String defaultErrorReport;
+  private static String defaultErrorReportWithTemplate;
+  private static String errorReportWithStackTraceAndTitle;
+  private static String systemLineSeparator = "\n";
+
+  @BeforeClass
+  @SneakyThrows
+  public static void setUpStatic() {
+    defaultErrorReport = IOUtils.resourceToString("/error_report_uri.txt", StandardCharsets.UTF_8).strip();
+    defaultErrorReportWithTemplate = IOUtils.resourceToString("/error_report_uri_with_template.txt",
+        StandardCharsets.UTF_8).strip();
+    errorReportWithStackTraceAndTitle = IOUtils.resourceToString("/error_report_uri_with_stacktrace_title.txt",
+        StandardCharsets.UTF_8).strip();
+  }
+
+  @Before
+  public void setUp() {
+    Whitebox.setInternalState(GitMacheteErrorReportSubmitter.class, errorReportSubmitterLogger);
+    Whitebox.setInternalState(SystemUtils.class, "OS_NAME", "Mock OS X");
+    Whitebox.setInternalState(SystemUtils.class, "OS_VERSION", "Hehe");
+
+    PowerMockito.stub(PowerMockito.method(System.class, "lineSeparator"))
+            .toReturn(systemLineSeparator);
+
+    PowerMockito.mockStatic(ApplicationInfo.class);
+    PowerMockito.mockStatic(PluginId.class);
+
+    val applicationInfo = PowerMockito.mock(ApplicationInfo.class);
+    PowerMockito.when(applicationInfo.getFullApplicationName()).thenReturn("mocked IntelliJ idea");
+    PowerMockito.stub(PowerMockito.method(ApplicationInfo.class, "getInstance"))
+        .toReturn(applicationInfo);
+  }
+
+  private IdeaLoggingEvent[] getMockEvents(String throwableText) {
+    val customThrowable = new Throwable() {
+      @Override
+      public void printStackTrace(PrintWriter s) {
+        s.print(throwableText);
+      }
+    };
+
+    return new IdeaLoggingEvent[]{
+        new IdeaLoggingEvent(/* message */ "", customThrowable)
+    };
+  }
+
+  private String invokeGetReportBody(GitMacheteErrorReportSubmitter errorReportSubmitter, IdeaLoggingEvent[] events)
+      throws Exception {
+    String reportBody = Whitebox.invokeMethod(errorReportSubmitter, "getReportBody", events, /* additionalInfo */ null);
+
+    Mockito
+        .verify(errorReportSubmitterLogger,
+            never().description("Reading the resource file with the bug template should NOT throw exceptions"))
+        .error(anyString(), (Throwable) any());
+
+    return reportBody;
+  }
+
+  @Test
+  public void shouldConstructUriWithTemplate() throws Exception {
+    val events = getMockEvents(/* throwableText */ "");
+    val reportBody = invokeGetReportBody(reportSubmitter, events);
```

Comment:
One more thing, to simplify the tests even more: what if you reworked `constructNewGitHubIssueUri` to sth like:
```java
  private URI constructNewGitHubIssueUri(IdeaLoggingEvent[] events, @Nullable String additionalInfo) throws URISyntaxException {
    String reportBody = getReportBody(events, additionalInfo);
    ....
```

Then AFAICS all the engineering around `getReportBody` won't even be needed in the test

Reply:
We still have to make sure no exception get's thrown inside getReportBody(), so most of the engineering with the logger will stay the same.

Reply:
Hmmm won't the test fail anyway if an exception is thrown inside `getReportBody`? 🤔

Reply:
No, as it's caught inside the method, LOG.error(...) is called inside the catch block to inform the outside world about the exception.